src/github.com/stellar/gateway/compliance/handlers/request_handler_tx_status.go

```diff
@@ -0,0 +1,87 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/stellar/gateway/protocols"
+	"github.com/stellar/gateway/server"
+	"github.com/stellar/go/protocols/compliance"
+)
+
+// HandlerTxStatus implements /tx_status endpoint
+func (rh *RequestHandler) HandlerTxStatus(w http.ResponseWriter, r *http.Request) {
+
+	txid := r.URL.Query().Get("id")
+	if txid == "" {
+		log.Info("unable to get query parameter")
+		server.Write(w, protocols.NewMissingParameter("id"))
+		return
+	}
+	response := compliance.TransactionStatusResponse{}
+
+	if rh.Config.Callbacks.TxStatus == "" {
+		response.Status = compliance.TransactionStatusUnknown
+	} else {
+		endpoint := fmt.Sprintf(
+			"%s?id=%s",
+			rh.Config.Callbacks.TxStatus,
+			txid,
+		)
+
+		_, err := url.Parse(endpoint)
+		if err != nil {
+			log.Error(err, "failed to parse tx status endpoint")
+			server.Write(w, protocols.InternalServerError)
+			return
+		}
```

Comment:
This will work in this case because `txid` doesn't contain special characters like `=`, `+` or `&` but the ideal solution would be something like:
```go
    u, err := url.Parse(endpoint)
 		if err != nil {
 			log.Error(err, "failed to parse tx status endpoint")
 			server.Write(w, protocols.InternalServerError)
 			return
 		}
    q := req.URL.Query()
    q.Add("id", txid)
    u.RawQuery = q.Encode()
    resp, err := rh.Client.Get(u.String())
```
This will automatically escape GET params' values.

Reply:
thanks for pointing this out. will change it

Reply:
You don't need:
```go
 		endpoint := fmt.Sprintf(
 			"%s?id=%s",
 			rh.Config.Callbacks.TxStatus,
 			txid,
 		)
```
anymore.